gunicorn/arbiter.py

```diff
@@ -519,7 +538,13 @@ def manage_workers(self):
 
     def spawn_worker(self):
         self.worker_age += 1
-        worker = self.worker_class(self.worker_age, self.pid, self.LISTENERS,
+        # self.LISTENERS is the list of default sockets that workers listen on.
+        listeners = copy.copy(self.LISTENERS)
+        # When self.WORKER_SOCKETS is populated, we are to assign one worker listener per worker class.
+        if self.WORKER_LISTENERS:
+            listeners.append(self.WORKER_LISTENERS[self.worker_address_index])
+            self.worker_address_index = (self.worker_address_index + 1) % len(self.WORKER_LISTENERS)
+        worker = self.worker_class(self.worker_age, self.pid, listeners,
```

Comment:
For this block of code, can it be simplified? .. I think this has the same result:

```python
def spawn_worker(self):
    # self.LISTENERS is the list of default sockets that workers listen on.
    # When self.WORKER_SOCKETS is populated, we are to assign one worker listener per worker class.
    if self.WORKER_LISTENERS:
        listeners = self.LISTENERS + [self.WORKER_LISTENERS[self.worker_age % len(self.WORKER_LISTENERS)]]
    else:
        listeners = self.LISTENERS
    self.worker_age += 1
    worker = self.worker_class(self.worker_age, self.pid, listeners,
```